dask/dataframe/io/parquet/fastparquet.py

```diff
@@ -299,18 +303,16 @@ def read_metadata(
         # if we have a list of files and gather_statistics=False
         if not parts:
             partsin = pf.row_groups
-            pf.fmd.key_value_metadata = None
+            if fast_metadata:
+                pf = (paths, gather_statistics)
         else:
-            pf = None
             partsin = parts
+            pf = None
         parts = []
-        for piece in partsin:
-            if pf is not None:
-                for col in piece.columns:
-                    col.meta_data.statistics = None
-                    col.meta_data.encoding_stats = None
```

Comment:
In the case that we pass a ParequetFile, it is still worth stripping all we can, no? Or did all row groups get removed anyway?

Reply:
Right - We can still strip this information (adding it back in)